### PR TITLE
LITE-18976: Add validation command to extension projects

### PR DESCRIPTION
**Pull request by jpdolz**

### CODE REVIEW

**Review by vlas — APPROVED** [review]
LGTM

File: connect/cli/plugins/project/constants.py [hunk 1]
```diff
@@ -2,3 +2,29 @@
...
+    'asset_change_request_validation': 'validate_asset_change_request',
+    'product_action_execution': 'execute_product_action',
+    'product_custom_event_processing': 'process_product_custom_event',
+    'tier_config_setup_request_processing': 'process_tier_config_setup_request',
+    'tier_config_change_request_processing': 'process_tier_config_change_request',
+    'tier_config_setup_request_validation': 'validate_tier_config_setup_request',
+    'tier_config_change_request_validation': 'validate_tier_config_change_request',
+}
+
+CAPABILITY_ALLOWED_STATS = [
```
Comment (ffaraoneim) [inline connect/cli/plugins/project/constants.py:23]: better STATUSES not STATS

File: connect/cli/plugins/project/extension_helpers.py [hunk 2]
```diff
@@ -41,3 +51,110 @@ def bootstrap_extension_project(data_dir: str):
...
+        return
+
+    raise ClickException('An Extension class can only have sync or async methods not a mix of both.')
+
+
+def _have_capabilities_proper_stats(capabilities):
+    errors = []
+    for capability, stats in capabilities.items():
+        if capability == 'product_action_execution' or capability == 'product_custom_event_processing':
+            if stats:
```
Comment (ffaraoneim) [inline connect/cli/plugins/project/extension_helpers.py:147]: In this case, we should ensure that the value of these two capabilities is an empty list. A null value can cause issues.

**Review by ffaraoneim — CHANGES_REQUESTED** [review]

File: connect/cli/plugins/project/constants.py (hunk 1 above)
Comment (jpdolz) [reply connect/cli/plugins/project/constants.py]: Done!

File: connect/cli/plugins/project/extension_helpers.py (hunk 2 above)
Comment (jpdolz) [reply connect/cli/plugins/project/extension_helpers.py]: Done!

**Review by ffaraoneim — APPROVED** [review]

**Pull request merged**